File: src/org/anibyl/slounik/SlounikAdapter.java
package org.anibyl.slounik;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import java.util.ArrayList;

/**
 * Adapter used to fill the list.
 * <p/>
 * Created by Usievaład Čorny on 26.02.2015 13:28.
 */
public class SlounikAdapter extends ArrayAdapter {
    private LayoutInflater inflater;
    private ArrayList<Article> list;

    public SlounikAdapter(Context context, int resource, int textViewResourceId, ArrayList<Article> list) {
        super(context, resource, textViewResourceId, list);

        inflater = LayoutInflater.from(context);
        this.list = list;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        ViewHolder holder;
        if (convertView == null) {
            convertView = inflater.inflate(R.layout.list_item, null);
            holder = new ViewHolder();
            holder.title = (TextView) convertView.findViewById(R.id.title);
            holder.description = (TextView) convertView.findViewById(R.id.description);
            holder.dicName = (TextView) convertView.findViewById(R.id.dicName);

            convertView.setTag(holder);
        } else {
            holder = (ViewHolder) convertView.getTag();
        }

        Article article = list.get(position);
        holder.title.setText(article.getTitle());
        holder.description.setText(article.getDescription());
        holder.dicName.setText(article.getDictionary());

        return convertView;
    }

    static class ViewHolder {
        TextView title;
        TextView description;
        TextView dicName;
    }
}

File: main/src/main/java/org/anibyl/slounik/SlounikAdapter.kt
package org.anibyl.slounik

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import org.anibyl.slounik.network.Article

import java.util.ArrayList

/**
 * Adapter used to fill the list.
 *
 * @author Usievaład Kimajeŭ
 * @created 26.02.2015
 */
class SlounikAdapter(context:Context, resource:Int, textViewResourceId:Int, private val list:ArrayList<Article>)
:ArrayAdapter<Article>(context, resource, textViewResourceId, list) {
	private val inflater:LayoutInflater

	init {
		inflater = LayoutInflater.from(context)
	}

	override fun getView(position:Int, convertView:View?, parent:ViewGroup):View {
		val holder:ViewHolder
		val view:View
		if (convertView == null) {
			view = inflater.inflate(R.layout.list_item, null)
			holder = ViewHolder(
				view.findViewById(R.id.list_item_title) as TextView,
				view.findViewById(R.id.list_item_description) as TextView,
				view.findViewById(R.id.list_item_dictionary) as TextView
			)

			view.tag = holder
		} else {
			view = convertView
			holder = convertView.tag as ViewHolder
		}

		val article = list[position]
		holder.title.text = article.title
		holder.description.text = article.description
		holder.dicName.text = article.dictionary

		return view
	}

	internal class ViewHolder(val title:TextView, val description:TextView, val dicName:TextView)
}
